sim/battle.ts

```diff
@@ -2790,6 +2797,12 @@ export class Battle {
 			this.queue.shift();
 			this.runAction(action);
 			if (this.requestState || this.ended) return;
+
+			if (this.gen >= 8 && this.turn > 0) {
+				// Dynamically updated speed mid-turn except during team selection.
+				this.updateSpeed();
+				this.sortQueue();
+			}
```

Comment:
This should be moved to the end of `runAction`.

Also, for performance reasons, you might as well just do this:

```
 if (this.gen >= 8 && this.queue.length && this.queue[0].choice === 'move') {
```